### PR TITLE
Fixed up ability to purge celery queues which use redispriorityasync

### PR DESCRIPTION
**Pull request by sbarman**
For @yorinasub17 

I fixed up the purge function.

I also fixed up some other parts of the transport that still used code from the list backend and added exceptions to parts of the code that are used in synchronous mode.

### CODE REVIEW

File: kombu_redis_priority/transport/redis_priority_async.py [hunk 1]
```diff
@@ -596,29 +592,20 @@ def _poll_error(self, type, **options):
...
-            raise Empty()
+        # This method is used with a synchronous Kombu channel, which is not currently supported.
+        # The old code is left here in case we want to support synchronous channels one day.
+        #
+        # with self.conn_or_acquire() as client:
+        #     for pri in self.priority_steps:
+        #         item = client.rpop(self._q_for_pri(queue, pri))
+        #         if item:
+        #             return loads(bytes_to_str(item))
+        #     raise Empty()
```
Comment (yorinasub17) [inline kombu_redis_priority/transport/redis_priority_async.py]: Maybe we should delete this and instead point to this PR?

File: kombu_redis_priority/transport/redis_priority_async.py [hunk 2]
```diff
@@ -658,17 +645,13 @@ def _delete(self, queue, exchange, routing_key, pattern, *args, **kwargs):
...
 
     def _has_queue(self, queue, **kwargs):
         with self.conn_or_acquire() as client:
-            with client.pipeline() as pipe:
-                for pri in self.priority_steps:
-                    pipe = pipe.exists(self._q_for_pri(queue, pri))
-                return any(pipe.execute())
+            if client.exists(queue):
+                return True
+            return False
```
Comment (yorinasub17) [inline kombu_redis_priority/transport/redis_priority_async.py]: should this just `return client.exists(queue)`?

File: tests/test_sortedset_transport.py [hunk 3]
```diff
@@ -95,3 +95,38 @@ def test_zrem_read(self):
...
+        self.assertEqual(len(raw_db), 0)
+
+        # put a blank message
+        self.channel._put('foo', {})
+        # verify queue is created
+        self.assertEqual(len(raw_db), 1)
+
+        num_msg = self.channel._purge('foo')
+        # verify that we removed one message and the key in Redis does not exist
+        self.assertEqual(num_msg, 1)
```
Comment (yorinasub17) [inline tests/test_sortedset_transport.py:112]: NIT: This isn't verifying if the key has been deleted. Either fix comment, or test.

**Review by yorinasub17 — APPROVED** [review]
LGTM except for nits and have verified it working in the celery environment:

```
(cipher) (ruby-1.9.3-p551) yoriy@Yoriyasus-MacBook-Pro:~/captricity/Cipher/cipher  (master %=)
~> celery -A cipher amqp queue.purge conversion-fast
-> connecting to redispriorityasync://localhost:6379/3.
-> connected.
Command # 1 (LLEN conversion-fast) of pipeline caused error: WRONGTYPE Operation against a key holding the wrong kind of value

(cipher) (ruby-1.9.3-p551) yoriy@Yoriyasus-MacBook-Pro:~/captricity/Cipher/cipher  (master %=)
~> pip install ~/captricity/kombu-redis-priority/
Processing /Users/yoriy/captricity/kombu-redis-priority
Requirement already satisfied: kombu in /Users/yoriy/.virtualenvs/cipher/lib/python2.7/site-packages (from kombu-redis-priority==0.1.1)
Requirement already satisfied: amqp<3.0,>=2.1.4 in /Users/yoriy/.virtualenvs/cipher/lib/python2.7/site-packages (from kombu->kombu-redis-priority==0.1.1)
Requirement already satisfied: vine>=1.1.3 in /Users/yoriy/.virtualenvs/cipher/lib/python2.7/site-packages (from amqp<3.0,>=2.1.4->kombu->kombu-redis-priority==0.1.1)
Building wheels for collected packages: kombu-redis-priority
  Running setup.py bdist_wheel for kombu-redis-priority ... done
  Stored in directory: /Users/yoriy/Library/Caches/pip/wheels/72/a8/34/e531885399bd140715977233e96c07be6fae4e4f145725136a
Successfully built kombu-redis-priority
Installing collected packages: kombu-redis-priority
  Found existing installation: kombu-redis-priority 0.1.0
    Uninstalling kombu-redis-priority-0.1.0:
      Successfully uninstalled kombu-redis-priority-0.1.0
Successfully installed kombu-redis-priority-0.1.1

(cipher) (ruby-1.9.3-p551) yoriy@Yoriyasus-MacBook-Pro:~/captricity/Cipher/cipher  (master %=)
~> celery -A cipher amqp queue.purge conversion-fast
-> connecting to redispriorityasync://localhost:6379/3.
-> connected.
ok. 0 messages deleted.
```

File: tests/test_sortedset_transport.py (hunk 3 above)
Comment (sbarman) [reply tests/test_sortedset_transport.py:112]: added the test

**Pull request merged**